external-effects/super-blur/CMakeLists.txt

```diff
@@ -0,0 +1,161 @@
+        # Include FFmpeg headers (needed for compile)
+        list(APPEND FF_INCLUDES ${${ffmpeg_comp}_INCLUDE_DIRS})
+        add_definitions(${${ffmpeg_comp}_DEFINITIONS})
+    endif()
+endforeach()
+list(REMOVE_DUPLICATES FF_INCLUDES)
+include_directories(${FF_INCLUDES})
+
+################# LIBOPENSHOT-AUDIO ###################
+# Find JUCE-based openshot Audio libraries
+FIND_PACKAGE(OpenShotAudio 0.1.8 REQUIRED)
+
+# Include Juce headers (needed for compile)
+include_directories(${LIBOPENSHOT_AUDIO_INCLUDE_DIRS})
+
+################# QT5 ###################
+# Find QT5 libraries
+foreach(qt_lib Qt5Widgets Qt5Core Qt5Gui Qt5Multimedia Qt5MultimediaWidgets)
+    find_package(${qt_lib} REQUIRED)
+    # Header files
+    list(APPEND QT_INCLUDES ${${qt_lib}_INCLUDE_DIRS})
+    # Compiler definitions
+    add_definitions(${${qt_lib}_DEFINITIONS})
+    # Other CFLAGS
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${${qt_lib}_EXECUTABLE_COMPILE_FLAGS}")
+endforeach()
+list(REMOVE_DUPLICATES QT_INCLUDES)
+include_directories(${QT_INCLUDES})
+
+# Manually moc Qt files
+qt5_wrap_cpp(MOC_FILES ${QT_HEADER_FILES})
+
+
+################# BLACKMAGIC DECKLINK ###################
+IF (ENABLE_BLACKMAGIC)
+	# Find BlackMagic DeckLinkAPI libraries
+	FIND_PACKAGE(BlackMagic)
+
+	IF (BLACKMAGIC_FOUND)
+		# Include Blackmagic headers (needed for compile)
+		include_directories(${BLACKMAGIC_INCLUDE_DIR})
+	ENDIF (BLACKMAGIC_FOUND)
+ENDIF (ENABLE_BLACKMAGIC)
+
+################### OPENMP #####################
+# Check for OpenMP (used for multi-core processing)
+FIND_PACKAGE(OpenMP)
+
+if (OPENMP_FOUND)
+   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS} ")
+endif(OPENMP_FOUND)
+
+################### ZEROMQ #####################
+# Find ZeroMQ library (used for socket communication & logging)
+FIND_PACKAGE(ZMQ REQUIRED)
+
+# Include ZeroMQ headers (needed for compile)
+include_directories(${ZMQ_INCLUDE_DIRS})
+
+################### RESVG #####################
+# Find resvg library (used for rendering svg files)
+FIND_PACKAGE(RESVG)
+
+# Include resvg headers (optional SVG library)
+if (RESVG_FOUND)
+	include_directories(${RESVG_INCLUDE_DIRS})
+endif(RESVG_FOUND)
+
+################### JSONCPP #####################
+# Include jsoncpp headers (needed for JSON parsing)
+if (USE_SYSTEM_JSONCPP)
+	find_package(JsonCpp REQUIRED)
+	include_directories(${JSONCPP_INCLUDE_DIRS})
+else()
+	include_directories("../../thirdparty/jsoncpp")
+endif(USE_SYSTEM_JSONCPP)
+
+SET ( EFFECT_SOURCE_FILES
+		SuperBlur.cpp)
+
+IF (NOT DISABLE_EXTERNAL_EFFECTS)
+	# Create shared openshot library
+	add_library(libeffectsuperblur SHARED
+			${EFFECT_SOURCE_FILES})
+
+	set(LIB_INSTALL_DIR libeffect${LIB_SUFFIX}) # determine correct lib folder
+
+	add_custom_command(
+			TARGET libeffectsuperblur
+			POST_BUILD
+			COMMAND ${CMAKE_COMMAND} -E copy
+			$<TARGET_FILE:libeffectsuperblur>
+			${PROJECT_SOURCE_DIR}/plugins/$<TARGET_FILE_NAME:libeffectsuperblur>
+	)
```

Comment:
Is this... copying the _shared library_ back into the source dir?!? Why? Is it just so the tests can load the effect plugin `.so`, later?

There's no need to pollute the source tree for that, the `CMakeLists.txt` for the unit tests can provide a compile-time definition for the _output_ location for the external plugin, in the build tree. (`${PROJECT_BINARY_DIR}/external-effects/super-blur/` in this case.) The tests can load it right from there, instead.

Reply:
Removed, fixed